eosetl/jobs/export_blocks_job.py

```diff
@@ -0,0 +1,105 @@
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+from eosetl.mappers.action_mapper import EosActionMapper
+from eosetl.mappers.block_mapper import EosBlockMapper
+from eosetl.mappers.transaction_mapper import EosTransactionMapper
+from eosetl.service.btc_service import EosService
+from blockchainetl.executors.batch_work_executor import BatchWorkExecutor
+from blockchainetl.jobs.base_job import BaseJob
+from blockchainetl.utils import validate_range
+
+
+# Exports blocks and transactions
+class ExportBlocksJob(BaseJob):
+    def __init__(
+            self,
+            start_block,
+            end_block,
+            batch_size,
+            eos_rpc,
+            max_workers,
+            item_exporter,
+            chain,
+            export_blocks=True,
+            export_transactions=True):
+        validate_range(start_block, end_block)
+        self.start_block = start_block
+        self.end_block = end_block
+
+        self.batch_work_executor = BatchWorkExecutor(batch_size, max_workers)
+        self.item_exporter = item_exporter
+
+        self.export_blocks = export_blocks
+        self.export_transactions = export_transactions
+        if not self.export_blocks and not self.export_transactions:
+            raise ValueError('At least one of export_blocks or export_transactions must be True')
+
+        self.eos_service = EosService(eos_rpc, chain)
+        self.block_mapper = EosBlockMapper()
+        self.transaction_mapper = EosTransactionMapper()
+        self.action_mapper = EosActionMapper()
+
+    def _start(self):
+        self.item_exporter.open()
+
+    def _export(self):
+        self.batch_work_executor.execute(
+            range(self.start_block, self.end_block + 1),
+            self._export_batch,
+            total_items=self.end_block - self.start_block + 1
+        )
+
+    def _export_batch(self, block_number_batch):
+        blocks = self.eos_service.get_blocks(block_number_batch, self.export_transactions)
+        for block in blocks:
+            self._export_block(block)
+            self._export_txs(block)
+
+    def _export_block(self, block):
+        if not self.export_blocks:
+            return
+
+        self.item_exporter.export_item(self.block_mapper.block_to_dict(block))
+
+    def _export_txs(self, block):
```

Comment:
Use transaction instead of tx for consistency with other parts of the code @Vasiliy-Bondarenko .

Reply:
done